src/store/ChartSettingStore.js

```diff
@@ -104,37 +104,67 @@ export default class ChartSettingStore {
         this.defaultLanguage = this.languages[0] || languageList[0];
     }
 
-   setSettings(settings) {
-       if (settings === undefined) { return; }
-       const { assetInformation, countdown, historical, language, position, isAutoScale, isHighestLowestMarkerEnabled, theme } = settings;
-       if (theme                        !== undefined) { this.setTheme(theme); }
-       if (position                     !== undefined) { this.setPosition(position); }
-       if (countdown                    !== undefined) { this.showCountdown(countdown); }
-       if (language                     !== undefined) { this.setLanguage(language); }
-       if (assetInformation             !== undefined) { this.setAssetInformation(assetInformation); }
-       if (historical                   !== undefined) { this.setHistorical(historical); }
-       if (isAutoScale                  !== undefined) { this.setAutoScale(isAutoScale); }
-       if (isHighestLowestMarkerEnabled !== undefined) { this.toggleHighestLowestMarker(isHighestLowestMarkerEnabled); }
-   }
-
-   saveSetting() {
-       if (this.onSettingsChange) {
-           this.onSettingsChange({
-               assetInformation            : this.assetInformation,
-               countdown                   : this.countdown,
-               historical                  : this.historical,
-               language                    : this.language.key,
-               position                    : this.position,
-               isAutoScale                 : this.isAutoScale,
-               isHighestLowestMarkerEnabled: this.isHighestLowestMarkerEnabled,
-               theme                       : this.theme,
-           });
-       }
-   }
+    @action.bound updateProps(activeLanguages) {
```

Comment:
@shayan-binary it should be like this
`@action.bound updateProps({ activeLanguages }) {`